packages/gatsby-theme-store/src/sdk/buyButton/useBuyButton.ts

```diff
@@ -73,6 +89,20 @@ export const useBuyButton = ({
       return
     }
 
+    const pixelEventItem = {
+      productId: product?.id,
+      productReferenceId: product?.productReference,
+      productName: product?.productName ?? productName,
+      brand: product?.brand,
+      categoryTree: product?.categoryTree,
+      price: seller.commercialOffer.price,
+      // TODO currencyCode,
+      quantity,
+      skuId: sku?.itemId,
+      skuName: sku?.name,
+      skuReferenceId: sku?.referenceId,
+    } as PixelItem
```

Comment:
Can you explain why this cast is needed?

Reply:
Because for now, the `product` property is optional. `undefined` would not be accepted by the `PixelItem` type, so I had to cast it since there's no problem with it being `undefined` for now. When I update all the stores with the `product` property, it'll be mandatory, and this cast will be also removed.